docs/08-resources.md

```diff
@@ -0,0 +1,143 @@
+---
+title: Assets
+path: /docs/assets
+---
+
+Almost every game is more than just boxes and circles. Excalibur provides an asset loading feature that can help you preload things like JSON files, images, and sounds.
+
+## Creating an asset loader
+
+When calling [[Engine.start]], you can optionally pass an asset [[Loader]]. This loader will contain a reference to any [[Loadable|"loadables"]] you want to load.
+
+```ts
+const loader = new ex.Loader([
+  /* add Loadables here */
+])
+```
+
+[[Loadable|Loadables]] are different kinds of assets such as textures, sounds, and generic resources.
+
+<docs-note>**Anytime** you call `game.start(loader)`, the game will pause and the engine will load assets. This means that you _do not have to load every asset at once_! Instead you may want to call `game.start(loader)` initially with core assets and then again when [initializing a Scene](/docs/scenes#initialization).</docs-note>
+
+## Using a web server
+
+The asset loader **only works with a web server** since it loads assets with [XHR](https://developer.mozilla.org/en-US/docs/Web/API/XMLHttpRequest). That means you cannot use the loader when double-clicking and running an HTML file in the browser. The browser throws errors that will prevent you from loading assets.
+
+The fastest way to serve a folder of files is by using the [serve](https://npmjs.org/package/serve) NPM package.
+
+```bash
+# Serve the current directory
+npx serve .
+
+# Serve a folder
+npx serve ./dist
+```
+
+If you are developing a game using Excalibur with Webpack, Parcel, or another bundler, these typically already come with dev servers for running your game.
+
+### Relative vs. absolute paths
+
+Given this directory structure:
+
+```
+/root
+  src/
+    game.js
+  assets/
+    textures/
+      map.png
+  index.html
+```
+
+And you serve from the `root` directory like this:
+
+```
+> cd root
+> npx serve .
+
+Now serving on http://localhost:3000/
+```
+
+The path to your assets doesn't matter as much because both absolute and relative paths will work:
+
+- `/assets/textures/map.png => HTTP 200 OK`
+- `assets/textures/map.png => HTTP 200 OK`
+
+But if you are serving under a sub-directory, like `http://localhost:3000/root/index.html` then the format of your paths matter:
+
+- `/assets/textures/map.png => HTTP 404 Not Found`
+- `assets/textures/map.png => HTTP 200 OK`
+
+The first path will fail to load as the absolute asset path would now be `/root/assets` and not `/assets`. Use a relative path to load assets _relative_ to the HTML file serving your game.
```

Comment:
Perhaps something can be mentioned about using [https://developer.mozilla.org/en-US/docs/Web/API/Node/baseURI](document.baseURI) combined with the [HTML base tag](https://developer.mozilla.org/en-US/docs/Web/HTML/Element/base). This allows for a reliable reference to the HTML file's base path.

Reply:
Didn't even think of that, good call!